modules/datastore/src/Service/PostImport.php

```diff
@@ -2,40 +2,252 @@
 
   /**
-   * The metastore resource mapper service.
+   * The logger service.
+   *
+   * @var \Psr\Log\LoggerInterface
+   */
+  protected LoggerInterface $logger;
+
+  /**
+   * The resource mapper.
+   *
+   * @var \Drupal\metastore\ResourceMapper
    */
   protected ResourceMapper $resourceMapper;
 
   /**
-   * Constructs a new PostImport object.
+   * The resource processor collector.
+   *
+   * @var \Drupal\datastore\Service\ResourceProcessorCollector
+   */
+  protected ResourceProcessorCollector $resourceProcessorCollector;
+
+  /**
+   * The data dictionary discovery interface.
    *
+   * @var \Drupal\metastore\DataDictionary\DataDictionaryDiscoveryInterface
+   */
+  protected DataDictionaryDiscoveryInterface $dataDictionaryDiscovery;
+
+  /**
+   * The reference lookup service.
+   *
+   * @var \Drupal\metastore\Reference\ReferenceLookup
+   */
+  protected ReferenceLookup $referenceLookup;
+
+  /**
+   * The post import result service.
+   *
+   * @var \Drupal\datastore\PostImportResult
+   */
+  protected PostImportResult $postImportResult;
+
+  /**
+   * The datastore service.
+   *
+   * @var \Drupal\datastore\DatastoreService
+   */
+  protected DatastoreService $datastoreService;
+
+  /**
+   * The database connection.
+   *
+   * @var \Drupal\Core\Database\Connection
+   */
+  protected Connection $connection;
+
+  /**
+   * Constructs a new PostImport service.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
+   *   The config factory service.
+   * @param \Psr\Log\LoggerInterface $logger
+   *   The logger service.
+   * @param \Drupal\metastore\ResourceMapper $resourceMapper
+   *   The resource mapper service.
+   * @param \Drupal\datastore\Service\ResourceProcessorCollector $resourceProcessorCollector
+   *   The resource processor collector service.
+   * @param \Drupal\metastore\DataDictionary\DataDictionaryDiscoveryInterface $dataDictionaryDiscovery
+   *   The data dictionary discovery interface.
+   * @param \Drupal\metastore\Reference\ReferenceLookup $referenceLookup
+   *   The reference lookup service.
+   * @param \Drupal\datastore\DatastoreService $datastoreService
+   *   The datastore service.
    * @param \Drupal\Core\Database\Connection $connection
    *   The database connection.
-   * @param \Drupal\metastore\ResourceMapper $resource_mapper
-   *   The metastore resource mapper service.
    */
   public function __construct(
+    ConfigFactoryInterface $configFactory,
+    LoggerInterface $logger,
+    ResourceMapper $resourceMapper,
+    ResourceProcessorCollector $resourceProcessorCollector,
+    DataDictionaryDiscoveryInterface $dataDictionaryDiscovery,
+    ReferenceLookup $referenceLookup,
+    DatastoreService $datastoreService,
     Connection $connection,
-    ResourceMapper $resource_mapper
-    ) {
+  ) {
+    $this->configFactory = $configFactory;
+    $this->logger = $logger;
+    $this->resourceMapper = $resourceMapper;
```

Comment:
Trying to think about how to get the number of dependencies down here. ResourceMapper is also part of DatastoreService. I think it would be safe to just make $datastoreService->resourceMapper public, since there are no public properties or setter methods on resourceMapper. Or we could add a getResourceMapper() method to $datastoreService.